src/languages/mathematica.js

```diff
@@ -6,61 +6,118 @@ Website: https://www.wolfram.com/mathematica/
 Category: scientific
 */
 
-export default function(hljs) {
+/*
+  The ./lib/mathematica.js file was created with Wolfram Mathematica 12.1.1 under Linux
+  with the following code:
+  $outputFile="/path/to/highlight.js/src/languages/lib/mathematica.js";
+  getStrippedContextNames[context_String]:=Block[
+    {
+      $ContextPath={context}
+    },
+    Union[Names@RegularExpression@StringJoin[context,"\\$?[A-Z]\\w*"]]
+  ];
+  $symbols=getStrippedContextNames["System`"];
+  Export[$outputFile,
+    With[
+      {
+        quoted="\""<>#<>"\""&/@$symbols
+      },
+      StringTemplate["export const SYSTEM_SYMBOLS = [\n`syms`\n];\n"][
+        <|"syms"->StringRiffle[Map["  " <> #&,quoted],",\n"]|>
+      ]
+    ],"String"]
+*/
+import * as Mathematica from './lib/mathematica.js';
+import * as regex from '../lib/regex.js';
+
+// @ts-ignore
+export default function(highlightJS) {
   /*
-  This dangerously looking beast of a regex was carefully assembled by Robert Jacobson.
-  See: https://wltools.github.io/LanguageSpec/Specification/Syntax/Number-representations/
+  This rather scary looking matching of Mathematica numbers is carefully explained by Robert Jacobson here:
+  https://wltools.github.io/LanguageSpec/Specification/Syntax/Number-representations/
    */
+  const base_re = /([2-9]|[1-2]\d|[3][0-5])\^\^/;
+  const base_digits_re = /(\w*\.\w+|\w+\.\w*|\w+)/;
+  const number_re = /(\d*\.\d+|\d+\.\d*|\d+)/;
+  const base_number_re = regex.either(regex.concat(base_re, base_digits_re), number_re);
+
+  const accuracy_re = /``[+-]?(\d*\.\d+|\d+\.\d*|\d+)/;
```

Comment:
Please use ALL_CAPS_CASE for constants like this.

Reply:
Hmmm... so there is a pattern here... you're using lowercase for the building blocks but then cap case for the modes?  Could you explain the thinking?  I just realized it was mixed and now I'm trying to understand it. :)

Reply:
I don't have a preference here. You guessed the idea correctly: lowercase for building blocks and upper case for the things that go into the final return value. I can make all uppercase if this is the convention for JS.

Reply:
> I can make all uppercase if this is the convention for JS.

I think it's a common convention and definitely one we typically follow.  I like it now that I get it, but sadly it's inconsistent with every other grammar so I think we could make them all caps.